src/Bicep.Core/Syntax/ImportDeclarationSyntax.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+using Bicep.Core.Navigation;
+using Bicep.Core.Parsing;
+using Bicep.Core.TypeSystem;
+
+namespace Bicep.Core.Syntax
+{
+    public class ImportDeclarationSyntax : StatementSyntax, ITopLevelNamedDeclarationSyntax
+    {
+        public ImportDeclarationSyntax(IEnumerable<SyntaxBase> leadingNodes, Token keyword, IdentifierSyntax aliasName, SyntaxBase fromKeyword, IdentifierSyntax providerName, SyntaxBase? config)
+            : base(leadingNodes)
+        {
+            AssertKeyword(keyword, nameof(keyword), LanguageConstants.ImportKeyword);
+            AssertSyntaxType(aliasName, nameof(aliasName), typeof(IdentifierSyntax));
+            AssertSyntaxType(fromKeyword, nameof(fromKeyword), typeof(Token), typeof(SkippedTriviaSyntax));
+            AssertSyntaxType(providerName, nameof(providerName), typeof(IdentifierSyntax));
+            AssertSyntaxType(config, nameof(config), typeof(ObjectSyntax), typeof(SkippedTriviaSyntax));
+
+            this.Keyword = keyword;
+            this.AliasName = aliasName;
+            this.FromKeyword = fromKeyword;
+            this.ProviderName = providerName;
+            this.Config = config;
+        }
+
+        public Token Keyword { get; }
+
+        public IdentifierSyntax AliasName { get; }
+
+        public SyntaxBase FromKeyword { get; }
+
+        public IdentifierSyntax ProviderName { get; }
```

Comment:
>Provider [](http://example.com/codeflow?start=32&length=8)

We really need to choose some good name for these things. `ProviderOfResourceTypesButNotResourceProvider`? ;)

Reply:
(we don't have to address it in this PR, of course)

Reply:
Yeah, totally. In the following syntax, what would you call `bar`?

`import foo from bar`

The way I've been thinking about it:
* `foo` is the `namespace alias`
* `bar` is the `namespace provider`

Reply:
Once you go into using it with functions or types, `foo` kind of becomes a "namespace" even if namespace alias is more accurate. I like "extension" for `bar`.